src/rpc/dial.rs

```diff
@@ -988,6 +1030,16 @@ async fn maybe_connect_via_webrtc(
                                             )),
                                         ));
                                     }
+                                    let mut caller_update_stats_inner =
+                                        caller_update_stats.lock().unwrap();
+                                    caller_update_stats_inner.count += 1;
+                                    let call_update_duration = call_update_start.elapsed();
+                                    if call_update_duration > caller_update_stats_inner.max_duration
+                                    {
+                                        caller_update_stats_inner.max_duration =
+                                            call_update_duration;
+                                    }
+                                    caller_update_stats_inner.total_duration = call_update_duration;
```

Comment:
maybe I'm not following correctly, but it looks to me that `total_duration` is updated with each ice candidate, and is updated to be the duration involved for that specific candidate. If that's correct, maybe we should change this to `current_duration`? `max_duration` to me implies the duration spent processing _all_ ice candidates.

Reply:
Ahh based on the reference to average duration in the `Display` implementation it looks like you are looking for the total duration to process all candidates. I _think_ that's not what we're getting here, since `total_duration` is updated based on `call_update_start` but `call_update_start` is redefined with each candidate. Should `call_update_start` be defined outside of `on_ice_candidate`? I'm thinking define it outside as a mutable `Option` type and then update it in `on_ice_candidate` only if it's a `None`, to make sure it stays stable within candidates while still accurately reflecting when you first started processing ice candidates.

Reply:
😮‍💨 entirely right; this should be `caller_update_stats_inner.total_duration += call_update_duration`. I _think_ that should fix things here, but let me know if you disagree.

Reply:
Ah yep, I think that would work! Didn't realize you could add durations, that's neat :)